components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/SynapsePolicyAggregator.java

```diff
@@ -237,15 +232,20 @@ private static List<String> renderPolicyMapping(List<OperationPolicy> policyList
         return renderedPolicyMappingList;
     }
 
-    private static String renderCustomBackendSequence(String sequenceName, String pathToArchive)
+    private static String renderCustomBackendSequence(String fileName, String pathToArchive, String sequenceName)
             throws APIManagementException {
         String policyDirectory = pathToArchive + File.separator + ImportExportConstants.CUSTOM_BACKEND_DIRECTORY;
-        String sequence = APIUtil.getCustomBackendSequenceFromFile(policyDirectory, sequenceName,
+        String sequence = APIUtil.getCustomBackendSequenceFromFile(policyDirectory, fileName,
                 APIConstants.SYNAPSE_POLICY_DEFINITION_EXTENSION_XML);
         if (sequence == null) {
             return null;
         }
-        return renderPolicyTemplate(sequence, new HashMap<>());
+        try {
+            sequence = updateSequenceName(sequence, sequenceName);
+        } catch (Exception ex) {
+            throw new APIManagementException("Error when updating the sequence name");
```

Comment:
Shall we improve the log to include more details and what haven't we throws the e